internal/providers/pluginfw/tfschema/struct_to_schema.go

```diff
@@ -172,18 +174,18 @@ func typeToSchema(v reflect.Value) map[string]AttributeBuilder {
 				elem := typeFieldType
 				sv := reflect.New(elem)
 				nestedScm := typeToSchema(sv)
-				scm[fieldName] = SingleNestedAttributeBuilder{
-					Attributes: nestedScm,
-					Optional:   isOptional,
-					Required:   !isOptional,
-					Computed:   isComputed,
+				scm_block[fieldName] = ListNestedBlockBuilder{
+					NestedObject: nestedScm,
+					Optional:     isOptional,
+					Required:     !isOptional,
+					Computed:     isComputed,
```

Comment:
TODO: Need to add validator for `MaxItem` = 1

Reply:
Looks like it's not very feasible right now given that we have updated the tfsdk structs directly to use slices for struct types.

The validation is done in the converter functions now.